kedro/framework/cli/project.py

```diff
@@ -234,3 +235,34 @@ def run(  # noqa: PLR0913
             pipeline_name=pipeline,
             namespace=namespace,
         )
+
+
+def _find_run_command(package_name: str) -> Any:  # pragma: no cover
+    from kedro.framework.cli.utils import (
+        KedroCliError,
+        load_entry_points,
+    )
+
+    try:
+        project_cli = importlib.import_module(f"{package_name}.cli")
+        # fail gracefully if cli.py does not exist
+    except ModuleNotFoundError as exc:
+        if f"{package_name}.cli" not in str(exc):
+            raise
+        plugins = load_entry_points("project")
+        _run = _find_run_command_in_plugins(plugins) if plugins else None
+        if _run:
+            # use run command from installed plugin if it exists
+            return _run
+        # use run command from `kedro.framework.cli.project`
+        return run
+    # fail badly if cli.py exists, but has no `cli` in it
+    if not hasattr(project_cli, "cli"):
+        raise KedroCliError(f"Cannot load commands from {package_name}.cli")
+    return project_cli.run
```

Comment:
Can we move this block right after `project_cli`? If the code hit the `except` logic it will return eagerly and never reach here.

Reply:
I'm not quite sure I follow, isn't it expected that if `cli.py` doesn't exist it goes to the `except` logic and never gets to the end?

Reply:
I read the code incorrectly.